pkg/util/https/server.go

```diff
@@ -43,23 +53,63 @@ type Server struct {
 // NewServer returns a Server instance.
 func NewServer(certFile, keyFile string) *Server {
 	mux := http.NewServeMux()
-	tls := &http.Server{
-		Addr:    ":8081",
-		Handler: mux,
-	}
 
 	wh := &Server{
 		Mux:      mux,
-		tls:      tls,
 		certFile: certFile,
 		keyFile:  keyFile,
 	}
+	wh.setupServer()
+
 	wh.Mux.HandleFunc("/", wh.defaultHandler)
 	wh.logger = runtime.NewLoggerWithType(wh)
 
 	return wh
 }
 
+func (s *Server) setupServer() {
+	s.tls = &http.Server{
+		Addr:    ":8081",
+		Handler: s.Mux,
+		TLSConfig: &cryptotls.Config{
+			GetCertificate: s.getCertificate,
+		},
+	}
+
+	// Start a goroutine to watch for certificate changes
+	go s.watchForCertificateChanges()
+}
+
+// getCertificate returns the current TLS certificate
+func (s *Server) getCertificate(hello *cryptotls.ClientHelloInfo) (*cryptotls.Certificate, error) {
+	s.CertMu.Lock()
+	defer s.CertMu.Unlock()
+	return s.Certs, nil
+}
+
+// watchForCertificateChanges watches for changes in the certificate files
+func (s *Server) watchForCertificateChanges() {
+	// Watch for changes in the tlsDir
+	cancelTLS, err := fswatch.Watch(s.logger, tlsDir, time.Second, func() {
+		// Load the new TLS certificate
+		tlsCert, err := cryptotls.LoadX509KeyPair(tlsDir+"server.crt", tlsDir+"server.key")
```

Comment:
I think the cert need to be fetched once outside the watcher so it can capture the initial value, similar as: https://github.com/googleforgames/agones/blob/main/cmd/allocator/main.go#L412

Reply:
Yes, You are right. I had added few logs and checked getCertificate() function is getting called but it has null value.